content/upcoming/enry.md

```diff
@@ -0,0 +1,183 @@
+
+The way the strategy chain works is as follows:
+* A strategy tries to get the language from a file.
+* It can result in zero or more languages detected.
+* If there's no language, call the next strategy.
+* If there are more than one languages, they are appended to a list of candidates and they are given to the next strategy.
+* If there is exactly one language, it is returned as the detected language and the process stops.
+
+ When the detection process falls through all the strategies without getting only one language as an outcome, it reaches the last step: `GetLanguagesByClassifier`. This strategy deserves taking a look at it.
+
+`GetLanguagesByClassifier` internally uses an object of the type:
+```
+type Classifier interface {
+    Classify(content []byte, candidates map[string]float64) (languages []string)
+}
+```
+
+This interface allows you to implement your own classifier and use it in the following functions:
+```
+func GetLanguageBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (language string, safe bool)
+func GetLanguagesBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (languages []string)
+```
+
+*enry*'s default classifier implementation is a [bayesian classifier](https://en.wikipedia.org/wiki/Bayes_classifier) which is like the one *linguist* uses. It assigns scores to the candidates regarding its probability, with the highest score assigned to the most likely candidate.
+
+By the way, strategies have a `GetLanguage-` version too that returns only a language, and boolean to indicate the sureness of this result. You can use this functions independently for whatever you want.
+
+If you want to customise *enry*'s default strategies and classifier to use your own strategies and classifiers, you can do so by assigning them to the following variables:
+```
+enry.DefaultStrategies = myStrategies
+enry.DefaultClassifier = myClassifier
+```
+From this point on, `GetLanguage` and `GetLanguages` will use your custom implementation.
+
+## How does enry know all about a lot of languages?
+*enry* uses the information about languages (filenames, interpreters, extensions... associated with each language) that *linguist* keeps on its project.
+
+Specifically, *enry* uses the following files from *linguist*:
+* [languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml)
+* [heuristics.rb](https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb)
+* [vendor.yml](https://github.com/github/linguist/blob/master/lib/linguist/vendor.yml)
+* [documentation.yml](https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml)
+
+These files are parsed to retrieve the necessary information. Then the source files in *enry*  that offer this information to the rest of the project are generated and encapsulated as an internal subpackage [data](https://github.com/src-d/enry/tree/master/data).
```

Comment:
May be there is a way to re-word this, without using `information` 3 times?

Something like `enry shares one "source of truth" for language identification with linguist`. `enry reads the following files...`  `These files are used to generate ..`, etc